compiler/src/dotty/tools/dotc/tastyreflect/CoreImpl.scala

```diff
@@ -24,6 +24,31 @@ trait CoreImpl extends scala.tasty.reflect.Core {
         type DefDef = tpd.DefDef
         type ValDef = tpd.ValDef
       type Term = tpd.Tree
+      val Term: TermCoreModuleImpl
+      trait TermCoreModuleImpl extends TermCoreModule {
+        type Ident = tpd.Ident
+        type Select = tpd.Select
+        type Literal = tpd.Literal
+        type This = tpd.This
+        type New = tpd.New
+        type NamedArg = tpd.NamedArg
+        type Apply = tpd.Apply
+        type TypeApply = tpd.TypeApply
+        type Super = tpd.Super
+        type Typed = tpd.Typed
+        type Assign = tpd.Assign
+        type Block = tpd.Block
+        type Lambda = tpd.Closure
+        type If = tpd.If
+        type Match = tpd.Match
+        type Try = tpd.Try
+        type Return = tpd.Return
+        type Repeated = tpd.SeqLiteral
+        type Inlined = tpd.Inlined
+        type SelectOuter = tpd.Select
```

Comment:
What is `SelectOuter`? I think maybe there is no need for constructor for `Inline`

Reply:
Select outer is a special case of a `Select` where it keeps track of the inlined scoped to avoid ambiguities.

Inline is used to keep track of the source file where the tree came from to have a correct position. If the macros would just drop them, every position in the resulting tree would be incorrect.